### PR TITLE
Implement Route53 Domains

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

File: resources/route53_domains.go [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+			{
+				Name:        "tags",
+				Description: "A list of tags",
+				Type:        schema.TypeJSON,
+				Resolver:    resolveRoute53DomainTags,
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:        "aws_route53_domain_admin_contact_extra_params",
```
Comment (roneli) [inline resources/route53_domains.go:361]: this relation can be a json, where name is the key and value is the value

File: resources/route53_domains.go [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+					},
+					{
+						Name:        "glue_ips",
+						Description: "Glue IP address of a name server entry",
+						Type:        schema.TypeStringArray,
+					},
+				},
+			},
+			{
+				Name:        "aws_route53_domain_registrant_contact_extra_params",
```
Comment (roneli) [inline resources/route53_domains.go:407]: same as admin extra params

File: resources/route53_domains.go [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+					},
+					{
+						Name:        "value",
+						Description: "The value that corresponds with the name of an extra parameter.  This member is required.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+			{
+				Name:        "aws_route53_domain_tech_contact_extra_params",
```
Comment (roneli) [inline resources/route53_domains.go:430]: same as admin extra params

**Review by roneli — CHANGES_REQUESTED** [review]
Looks Good, A few minor changes required, would also add e2e tests if possible + update changelog.

File: resources/route53_domains.go (hunk 1 above)
Comment (irmatov) [reply resources/route53_domains.go:361]: Done

File: resources/route53_domains.go (hunk 2 above)
Comment (irmatov) [reply resources/route53_domains.go:407]: Done

File: resources/route53_domains.go (hunk 3 above)
Comment (irmatov) [reply resources/route53_domains.go:430]: Done

**Comment by irmatov** [comment]
Changelog updated. Currently I don't see a way to provision this resource using Terraform. I suppose this resource actually registers the domain, so it's hardly testable.

File: resources/route53_domains.go [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/route53domains"
+	"github.com/aws/aws-sdk-go-v2/service/route53domains/types"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func Route53Domains() *schema.Table {
+	return &schema.Table{
```
Comment (roneli) [inline resources/route53_domains.go:16]: Missing ignore error

File: resources/route53_domains.go [hunk 5]
```diff
@@ -0,0 +1,492 @@
...
+		Name:         "aws_route53_domains",
+		Description:  "The domain names registered with Amazon Route 53.",
+		Resolver:     fetchRoute53Domains,
+		Multiplex:    client.AccountRegionMultiplex,
```
Comment (roneli) [inline resources/route53_domains.go:20]: If I am not incorrect only us-east-1 works on route53domains service. https://github.com/aws/aws-cli/issues/1354 and see https://docs.aws.amazon.com/general/latest/gr/r53.html

**Review by roneli — CHANGES_REQUESTED** [review]
Found minor issues in route53

File: resources/route53_domains.go (hunk 4 above)
Comment (irmatov) [reply resources/route53_domains.go:16]: Added:

		IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,

Also added PK on this and relation table

File: resources/route53_domains.go (hunk 5 above)
Comment (irmatov) [reply resources/route53_domains.go]: Good catch, thanks!

Updated to use only the default region.

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**